spectree/spec.py

```diff
@@ -190,6 +190,11 @@ def _generate_spec(self):
         }
         return spec
 
+    def _move_schema_reference(self, reference: str) -> str:
+        if '/definitions' in reference:
+            return f"#/components/schemas/{reference.split('/definitions/')[-1]}"
```

Comment:
Is it worth us adding nested definitions to the main definition list? Otherwise is our openapi spec going to be re-defining the same schema multiple times?

Reply:
The code maintains a flat dict of schemas and replaces rather than duplicates when it hits the same object again.  I think that this should be looked at again - what's really needed is a bit of proper tree traversal - but I'd rather not do that now...

Reply:
Gah! I totally missed that this is working right, I was still thinking about the upstream version. Shows how well I read the code :grimacing: